test/src/API/projects/bind.test.js

```diff
@@ -0,0 +1,134 @@
+        it('returns 202 when starting the bind process', async function() {
+            this.timeout(testTimeout.med);
+
+            const res = await projectService.bindStart({
+                name: projectName,
+                language: 'nodejs',
+                projectType: 'nodejs',
+                path: pathToLocalRepo,
+                creationTime: Date.now(),
+            });
+
+            res.should.have.status(202);
+            res.should.satisfyApiSpec;
+
+            projectID = res.body.projectID;
+        });
+        it('returns 200 for each file successfully uploaded to PFE', async function() {
+            this.timeout(testTimeout.med);
+
+            const responses = await projectService.uploadFiles(projectID, pathToLocalRepo);
+
+            responses.forEach(res => {
+                res.should.have.status(200);
+            });
+            const pathToUploadedFile = path.join('codewind-workspace', 'cw-temp', projectName, 'package.json');
+            const uploadedFileExists = await containerService.fileExists(pathToUploadedFile);
+            uploadedFileExists.should.be.true;
+        });
+        it('returns 200 when ending the bind process', async function() {
+            const res = await projectService.bindEnd(projectID);
+            res.should.have.status(200);
+            res.should.satisfyApiSpec;
+        });
+        it('returns 409 when trying to bind a project that is already bound', async function() {
+            const originalNumProjects = await projectService.countProjects();
+            await projectService.bindStart({
+                name: projectName,
+                language: 'nodejs',
+                projectType: 'nodejs',
+                path: pathToLocalRepo,
+            });
+            const finalNumProjects = await projectService.countProjects();
+            finalNumProjects.should.equal(originalNumProjects);
+        });
+    });
+    describe('POST /bind/start', () => {
+        describe('Failure Cases', () => {
+            it('returns 400 if projectName is invalid', async function() {
+                this.timeout(testTimeout.short);
+                const res = await projectService.bindStart({
+                    name: '<',
+                    language: 'nodejs',
+                    projectType: 'nodejs',
+                });
+                res.should.have.status(400);
+            });
+            it('returns 400 if projectType is invalid', async function() {
+                this.timeout(testTimeout.short);
+                const res = await projectService.bindStart({
+                    name: projectName,
+                    language: 'nodejs',
+                    projectType: 'invalid',
+                });
+                res.should.have.status(400);
+            });
+        });
+    });
+    describe('PUT /bind/upload', () => {
+        describe('Failure Cases', () => {
+            it('returns 400 for a project that does not exist', async function() {
```

Comment:
Being really picky but you check for `404` at the end of this test. (And the next one for bind end.)